content/blog/2020-08-10-darktable-3.2/2020-08-10-darktable-3.2.md

```diff
@@ -0,0 +1,452 @@
+
+## Performance improvements and important bug fixes
+
+A complete overhaul of the processing pipeline has been done by Pascal Obry, for a more robust management of the new pipeline integrated in 3.0. This overhaul makes the pipeline both more stable and efficient and fixes several bugs.
+
+Image modification detection has been improved. This mainly avoids the need to recalculate the display of an image that has not been altered, for example by having been opened in a darkroom without any modification. The modification of XMPs and thumbnails is more reliable and only impacted in the event of a real modification.
+
+Many other performance improvements have been integrated, especially in the processing modules. The work is ongoing and more improvements will be made in version 3.4. The next stable release is coming together with significant performance improvements.
+
+darktable 3.2 also introduces an automatic database maintenance option (configurable in the preferences), allowing for a healthier database to be maintained over time.
+
+Finally, with the various code improvements, many similar parts have been unified, allowing later a faster and easier integration of new modules and a much better consistency on similar parts (for example, what has been done on pickers).
+Test tools have also been added, making it easier for developers to test new additions and ensure backward compatibility and stability, especially when rewriting a module for better performance.
+
+And of course, many bugs have been fixed, making this 3.2 the most stable version 3 to date. If you are already using a 3.0.x version, I can recommend upgrading to version 3.2, if only for its superior stability.
+
+## Some new options in preferences window
+
+Apart from the new workflows and the few preferences already presented, you will find some new options that may be very useful. Here are the main ones:
+
+- In lighttable, you can sort film roll by number or by folder. Following the redesign of lighttable, you can also define different size categories (via size category separators) which will allow you to define a different overlay mode according to the size of the thumbnails. The size is specified in pixels and the thumbnail size is displayed in the overlay mode popover menu shown above.
+
+- In _darkroom_, it is now possible to reduce the resolution of the navigation preview, reducing CPU usage and possibly improving performance.
+**_Warning :_** as described in the tooltip of this option, this can slightly reduce the guided mask accuracy.
+
+- In miscellaneous tab, there are new options allow to set shortcuts if multiple instances are used.
```

Comment:
"In **the** miscellaneous tab there are new options to **control how keyboard shortcuts are applied where multiple module instances are used**"